Restructure project

imretoth-ess opened the pull request:
This PR adds repository, API, and service packages, and it splits up the controller code into controller and services. The PR will later be followed by one or more PRs to move business logic from the controller layer to the service layer.

This is a preparatory step in order to convert the HTTP API into a RESTful (v2) API. Early outlines for this work can be seen in e.g. https://github.com/ChannelFinder/ChannelFinderService/pull/106#issuecomment-1731079804.

imretoth-ess committed:
Move service classes

- Create package for service classes and move them there
- Adjust scopes for (service) functions to not to break current functionality

imretoth-ess committed:
Move repository classes

- Create package for repository classes and move them there

imretoth-ess committed:
Move configuration  classes

- Create package for configuration classes and move them there
- Adjust class name to represent it's function

imretoth-ess committed:
Move controller classes

-  Create package for controller classes and move them there

imretoth-ess committed:
Move util classes

- Create package for util classes and move them there

imretoth-ess committed:
Extract contrroller-level functions into interfaces

- Extract endpoint functions into interfaces
- Move endpoint (swagger) documentation into interfaces
- Remove unused imports from controller classes

imretoth-ess committed:
Refactor archiver packages

- Move classes into corresponding packages

imretoth-ess committed:
Rename controller classes

- Rename controller classes
- Rename controller interfaces

imretoth-ess committed:
Fix linting

- Some classes had incorrect linting that has been fixed

anderslindho reviewed (commented):
I think this looks good, but I'll defer approval to the next one in line to review this.

jacomago commented on src/main/java/org/phoebus/channelfinder/Application.java:
```diff
@@ -43,7 +43,7 @@
 public class Application implements ApplicationRunner {
 
-  static final Logger logger = Logger.getLogger(Application.class.getName());
+  public static final Logger logger = Logger.getLogger(Application.class.getName());
```
Not your fault, but we shouldn't have public loggers. They should be all private. Think you can fix this one?

imretoth-ess replied:
Sure, I will take a look at that!

jacomago commented on src/main/java/org/phoebus/channelfinder/service/ArchiverClient.java:
```diff
@@ -1,4 +1,4 @@
-package org.phoebus.channelfinder.processors.aa;
+package org.phoebus.channelfinder.service;
```
maybe make a sub external package for this class.

jacomago commented on src/main/java/org/phoebus/channelfinder/rest/controller/PropertyController.java:
```diff
@@ -647,29 +454,7 @@ public void remove(@PathVariable("propertyName") String propertyName) {
-  @DeleteMapping("/{propertyName}/{channelName}")
+  @Override
   public void removeSingle(
       @PathVariable("propertyName") final String propertyName,
```
Should we remove the pathvariable annotations now they are in the interface?

imretoth-ess replied:
Good catch, I will do that!

jacomago commented on src/main/java/org/phoebus/channelfinder/rest/controller/ChannelScrollController.java:
```diff
@@ -32,68 +28,32 @@
-import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
```
Should we get rid of the requestparam annotation as in the interface now?

jacomago reviewed (commented):
Couple of other things to do

imretoth-ess committed:
Adjust controllers

- Remove swagger related annotations from the controllers (as the documentation layer is moved to the interface level)

imretoth-ess committed:
Adjust scopes

- Wind down scope for loggers

imretoth-ess committed:
Create extra package

- Create external sub package for ArchiverClient

simon-ess reviewed (dismissed):
I might suggest squashing a few of the commits (e.g. when you rename the interfaces separately from creating them), but other than that LGTM.

anderslindho reviewed (commented):
Just noticed that you have a typo in a directory name, @imretoth-ess : `respository` -> `repository`

imretoth-ess commented:
> Just noticed that you have a typo in a directory name, @imretoth-ess : `respository` -> `repository`

Oh, good catch! I will fix that!

imretoth-ess committed:
Fix typo in package name

- Fix type in 'repository' package name

shroffk commented:
Is there an issue ticket or something with an overview of the changes and the need for it/problem being resolved?

anderslindho commented:
@shroffk the "epic" we're working towards here is enabling the v2 api

jacomago commented:
Ok, test was just flaky.

shroffk commented:
can you link the "epic" or something that describes the issues/tasks/motivations?

anderslindho commented:
@shroffk our ticketing system is not accessible from outside of ESS

@imretoth-ess can you please update your PR description/top level comment with this content (feel free to adjust as you see fit):
"""
This PR adds repository, API, and service packages, and it splits up the controller code into controller and services. The PR will later be followed by one or more PRs to move business logic from the controller layer to the service layer.

This is a preparatory step in order to convert the HTTP API into a RESTful (v2) API. Early outlines for this work can be seen in e.g. https://github.com/ChannelFinder/ChannelFinderService/pull/106#issuecomment-1731079804.
"""

@shroffk is the above clarification sufficient?
